src/main/java/org/apache/mesos/specification/TaskTypeSpecification.java

```diff
@@ -1,8 +1,16 @@
 package org.apache.mesos.specification;
 
+import org.apache.mesos.Protos;
+
+import java.util.Collection;
+
 /**
- * A TaskTypeSpecification describes the count of a particular TaskSpecification which should be launched.
+ * A TaskTypeSpecification describes a Type of Task and how many should be launched.
  */
-public interface TaskTypeSpecification extends TaskSpecification {
+public interface TaskTypeSpecification {
```

Comment:
I'm picturing someone accomplishing per-task handling via the construction of their task specification. Something like this:

```
public CustomTask implements Task {
  /** My custom constructor which does any per-task customization */
  public CustomTask(int index) {
    this.command = MyCustomLogic.getCommandFor(index);
  }
  [...]
  public String getCommand() {
    return this.command;
  }
}
```

Then the developer is free to make any per-instance customizations via that construction. Would this make sense, or is there something else going on here?